File: internal/screens/fixtures.go
package screens

import (
	"strings"

	"github.com/jashkarangiya/valo-tui/internal/data"
	"github.com/jashkarangiya/valo-tui/internal/styles"
)

// Fixtures is the [f] event sub-page: upcoming matches grouped by phase.
type Fixtures struct {
	w, h int
	body string
}

func NewFixtures(w, h int) Fixtures { return Fixtures{w: w, h: h} }

func (s *Fixtures) SetSize(w, h int) { s.w, s.h = w, h }

func (s *Fixtures) Load(eventID int, eventName string) {
	var b strings.Builder
	b.WriteString(title("fixtures") + "\n\n")
	if eventID == 0 {
		b.WriteString(hint("select an event first"))
		s.body = b.String()
		return
	}
	var upcoming []data.MatchCard
	for _, m := range data.EventMatchCards(eventID, eventName) {
		if m.Status == "upcoming" {
			upcoming = append(upcoming, m)
		}
	}
	if len(upcoming) == 0 {
		b.WriteString(hint("nothing scheduled for this event"))
		s.body = b.String()
		return
	}

	// Group by phase, preserving first-seen order.
	var phases []string
	seen := map[string]bool{}
	for _, m := range upcoming {
		ph := m.Phase
		if ph == "" {
			ph = "scheduled"
		}
		if !seen[ph] {
			seen[ph] = true
			phases = append(phases, ph)
		}
	}
	for _, ph := range phases {
		b.WriteString("\n" + accentB(ph) + "\n")
		for _, m := range upcoming {
			mp := m.Phase
			if mp == "" {
				mp = "scheduled"
			}
			if mp != ph {
				continue
			}
			when := m.Time
			if when == "" {
				when = m.Date
			}
			if when == "" {
				when = "soon"
			}
			card := textB(m.Team1.Name+"  vs  "+m.Team2.Name) + "   " + accent("·") + "  " + text(when)
			b.WriteString(styles.Card.Render(card) + "\n")
		}
	}
	s.body = strings.TrimRight(b.String(), "\n")
}

func (s Fixtures) View() string { return s.body }

File: valo_tui/screens/fixtures.py
"""[f] fixtures — what's still to come in the focused event.

Event-scoped upcoming matches as bento cards, grouped by phase so the bracket
rounds read in order."""

from __future__ import annotations

from textual.app import ComposeResult
from textual.containers import VerticalScroll
from textual.widgets import Label, Static

from ..data import cache
from .widgets import ACCENT, MUTED, TEXT


class FixturesView(VerticalScroll):
    can_focus = True

    def compose(self) -> ComposeResult:
        yield Label("fixtures", classes="page-title")
        yield VerticalScroll(id="fixtures-cards")

    def load_data(self) -> None:
        cards = self.query_one("#fixtures-cards", VerticalScroll)
        cards.remove_children()
        eid = getattr(self.app, "current_event_id", None)
        if eid is None:
            cards.mount(Label("select an event first", classes="hint"))
            return

        matches = [
            m for m in cache.event_match_cards(eid, self.app.event_name)
            if m.status == "upcoming"
        ]
        if not matches:
            cards.mount(Label("nothing scheduled for this event", classes="hint"))
            return

        # Group by phase, preserving first-seen order.
        phases: list[str] = []
        for m in matches:
            ph = m.phase or "scheduled"
            if ph not in phases:
                phases.append(ph)
        for ph in phases:
            cards.mount(Label(ph, classes="sched-section"))
            for m in [x for x in matches if (x.phase or "scheduled") == ph]:
                cards.mount(self._card(m))

    def _card(self, m) -> Static:
        when = m.time or m.date or "soon"
        title = (
            f"[bold {TEXT}]{m.team1.name}  vs  {m.team2.name}[/]"
            f"   [{ACCENT}]·[/]  [{TEXT}]{when}[/]"
        )
        return Static(title, classes="card")
